src/tabnineChatWidget/handlers/getEditorContextHandler.ts

```diff
@@ -58,6 +67,58 @@ export async function getEditorContext(): Promise<EditorContextResponse> {
   };
 }
 
+async function getSelectedCode(
+  editor: vscode.TextEditor,
+  request?: EditorContextRequest
+): Promise<string | undefined> {
+  const selectedCode = editor.document.getText(editor.selection);
+  if ((selectedCode && selectedCode !== "") || !request?.userQuery) {
+    return selectedCode;
+  }
+
+  const wordsInQuery = request.userQuery.split(/\s+/gm).slice(0, 20);
+  if (!wordsInQuery?.length) return undefined;
+  try {
+    return await executeTimeout(
+      async () => findUserQuerySymbolInCurrentFile(wordsInQuery, editor),
+      1000
+    );
+  } catch (e) {
+    Logger.debug(
+      `failed to fetch symbols for selected code: ${(e as Error).message}`
+    );
+    return undefined;
+  }
+}
+
+async function findUserQuerySymbolInCurrentFile(
+  wordsInQuery: string[],
+  editor: vscode.TextEditor
+): Promise<string | undefined> {
+  const allSymbols = await Promise.all(
+    wordsInQuery.map((word) =>
+      resolveSymbols({ symbol: word })
+        .then((symbols) =>
+          symbols?.filter(
+            (symbol) =>
+              symbol?.location?.uri.fsPath === editor.document.uri.fsPath
+          )
+        )
+        .catch((e) => {
+          Logger.debug(
+            `failed to fetch symbols for '${word}': ${(e as Error).message}`
+          );
+          return undefined;
+        })
+    )
+  );
+  const firstSymbol = allSymbols.find((s) => s?.length)?.find((s) => !!s);
```

Comment:
can you elaborate? why only the first symbol ?

Reply:
the selected code should be only one paragraph, so I figured lets go with the first occurrence of a symbol from the file.
What to you have in mind? to join all the occurrences?

Reply:
yep, tough to join them all... but we can start only with the first one